src/Compilers/Core/Portable/InternalUtilities/ExceptionUtilities.cs

```diff
@@ -19,7 +19,12 @@ internal static class ExceptionUtilities
         internal static Exception UnexpectedValue(object? o)
         {
             string output = string.Format("Unexpected value '{0}' of type '{1}'", o, (o != null) ? o.GetType().FullName : "<unknown>");
-            Debug.Assert(false, output);
+#if DEBUG
+            if (Debugger.IsAttached)
+            {
+                Debug.Assert(false, output);
+            }
+#endif
```

Comment:
@tmat This is causing a problem because we expect the ThrowingTraceListener to be active, but it's not taking effect in .NET Core scenarios so we failfast instead.

Reply:
This doesn't seem like the right solution. We have `Debug.Assert` in many places.

Reply:
Any suggestions for how to proceed here?

Reply:
Do we know why it doesn't work on Core? For now, perhaps we can skip fail-fast on Core.

Reply:
> Do we know why it doesn't work on Core?

`Debug.Assert` doesn't use trace listeners on .NET Core.

> For now, perhaps we can skip fail-fast on Core.

I have not found a way to change this behavior without private reflection, and even if we go that route I'm not sure how to do it from a single location in the xunit startup process.

Reply:
>  I'm not sure how to do it from a single location in the xunit startup process. 

I'm not advocating for this as a solution to the current problem. But keep in mind that ~1 months down the road this will actually be solvable with module initializers (having a singel init point for all xunit tests).

Reply:
As @tmat pointed out, we have `Debug.Assert` all over the code base. I don't see a particular reason to special case this instance over the hundreds of other ones we have. 

If we feel strongly that we should be able to control this we should start a dialog with the runtime team

Reply:
I discussed this with @stephentoub , and it looks like we can still use `ThrowingTraceListener`, but we have to add it explicitly:

```csharp
Trace.Listeners.Clear();
Trace.Listeners.Add(new ThrowingTraceListener());
```

Reply:
K. That pattern + module initailizers on .NET Core would seem to essentially cover this issue for us then.

Reply:
What release is the first to support module initializers? Are we already able to add one for the test helper assembly?

Reply:
The feature has been available in the runtime since 1.0 I believe. The language support will come in C# 9.0. The feature branch should get merged into the main branch sometime next week.

Reply:
This has been updated to use a module initializer to install the trace listener. There are two resulting impacts:

1. Visual Basic test projects targeting .NET Core cannot use `Debug.Assert` in any location where assertion might fail. Doing so will cause a `FailFast`. Workaround: if/when this occurs, make the test class extend `TestBase` to force module initialization via inheritance.
2. This PR is now blocked on using the `preview` language version (#44485).